lindeck-frontend/src/views/boody/deckPage/DeckPage.jsx

```diff
@@ -0,0 +1,128 @@
+import 'react-grid-layout/css/styles.css';
+import React, {Component} from 'react';
+import ReactGridLayout from "react-grid-layout";
+import ReactResizeDetector from 'react-resize-detector';
+import MovingDeckEditButton from "./deckEditMenu/DeckEditMenu";
+import DeckEditDialog from "./editDeckDialog/DeckEditDialog";
+import Card from "./card/Card";
+import GS from "../../../common/classes/GlobalStorage";
+import {Redirect} from "@reach/router";
+
+
+class deckPage extends Component {
+    state = {
+        deck: GS.getDeckFromUsernameDeckname(this.props.username, this.props.deckname),
+        editDeckDialogOpened: false,
+        gridWidth: 800,
+    }
+
+    addCard = (card) => {
+        let newDeck = this.state.deck
+        newDeck.addCard(card)
+        GS.saveDeck(newDeck)
+        this.setState({deck: newDeck})
+    }
+
+    resetLayout = () => {
+        let newDeck = this.state.deck
+        newDeck.resetLayout()
+        GS.saveDeck(newDeck)
+        this.setState({deck: newDeck})
+    }
+
+    clearAll = () => {
+        let newDeck = this.state.deck
+        newDeck.clear()
+        GS.saveDeck(newDeck)
+        this.setState({deck: newDeck})
+    }
+
+    duplicateCard = (card) => {
+        this.state.deck.duplicateCard(card)  // will  call onLayoutChange, dntknw wy but it is
+    }
+
+    deleteCard = (id) => {
+        this.state.deck.deleteCard(id) // mb because of moving deck edit menu (if it is open, layoutChange does not calling)
+    }
+
+    onLayoutChange(layout) {
+        let newDeck = this.state.deck
+        newDeck.layout = layout
+        GS.saveDeck(newDeck)
+        this.setState({deck: newDeck});
+    }
+
+    changeCardAnswer = (text, id) => {
+        this.state.deck.getCard(id).answer = text
+        GS.saveDeck(this.state.deck)
+    }
+
+    changeCardVerdict = (verdict, id) => {
+        this.state.deck.getCard(id).verdict = verdict
+        GS.saveDeck(this.state.deck)
+        this.setState({deck: this.state.deck});
+    }
+
+    changeCardText = (text, id, isFlipped) => {
+        if (isFlipped) {
+            this.state.deck.getCard(id).secondfield = text
+        } else {
+            this.state.deck.getCard(id).textfield = text
+        }
+        GS.saveDeck(this.state.deck)
+    }
+
+    openEditDeckDialog = (open) => {
+        this.setState({editDeckDialogOpened: open})
+    }
+
+    render() {
+        if (!this.state.deck) return <Redirect to="/nonfound" noThrow/>;
+        return <>
+            <MovingDeckEditButton
+                openEditDeckDialog={this.openEditDeckDialog} addCard={this.addCard}
+                resetLayout={this.resetLayout} clearAll={this.clearAll}>
+                <div className="backGround">
+                    <ReactResizeDetector handleWidth>
+                        {({width}) => <ReactGridLayout
+                            className="layout" cols={this.state.deck.cols}
+                            rowHeight={this.state.deck.rowHeight} width={width || 0}
```

Comment:
Default width is zero?

Reply:
yep. On the first render, ReactResizeDetector does not know about his child's size, so he pushes undefined to the width and it is an error.
So on the first render width equal to zero.